src/lib/logging/Logger.js

```diff
@@ -0,0 +1,423 @@
+     * @return {void}
+     */
+    info(...args) {
+        const message = arrayToString(args);
+        this.commitMessage(LOG_TYPES.info, message);
+    }
+
+    /**
+     * Log a warning message to the logger, and commit to the cache.
+     *
+     * @public
+     * @param {*} args - Arguments to be printed to the log and cache.
+     * @return {void}
+     */
+    warn(...args) {
+        const message = arrayToString(args);
+        this.commitMessage(LOG_TYPES.warning, message);
+    }
+
+    /**
+     * Log an error message to the logger, and commit to the cache.
+     *
+     * @public
+     * @param {*} args - Arguments to be printed to the log and cache.
+     * @return {void}
+     */
+    error(...args) {
+        const message = arrayToString(args);
+        this.commitMessage(LOG_TYPES.error, message);
+    }
+
+    /**
+     * Log a metric message to the logger, and commit to the cache.
+     *
+     * @public
+     * @param {string} eventName - Name associated with a specific metric.
+     * @param {*} value - Value of the metric
+     * @return {void}
+     */
+    metric(eventName, value) {
+        this.commitMessage(LOG_TYPES.metric, {
+            eventName,
+            value
+        });
+    }
+
+    /**
+     * Clear out the cache.
+     *
+     * @public
+     * @return {void}
+     */
+    clearCache() {
+        this.cache.clear();
+    }
+
+    /**
+     * Sets the current level of messages to be logged to the console.
+     *
+     * @public
+     * @param {CONSOLE_LEVELS|string} level - The level of logging to log to console.
+     * @param {boolean} [persist] - Whether or not to persist across sessions.
+     * @return {void}
+     */
+    setConsoleLevel(level, persist = false) {
+        this.logger.setLevel(level, persist);
+    }
+
+    /**
+     * Get logs from the cache.
+     *
+     * @public
+     * @param {LOG_TYPES|LOG_TYPES[]} [type] - Type of logs to get. If empty, will get all logs.
+     * If a list is given, will get each entry specified.
+     * @return {Object} The cache entry(ies) requested.
+     */
+    getLogs(type) {
+        const logs = {};
+
+        // eslint-disable-next-line require-jsdoc
+        const addGroupToLogs = (msgType) => {
+            const logType = LOG_TYPES[msgType];
+            logs[logType] = this.cache.getGroup(logType);
+        };
+
+        if (!type) {
+            Object.keys(LOG_TYPES).forEach(addGroupToLogs);
+        } else if (Array.isArray(type)) {
+            type.forEach(addGroupToLogs);
+        } else {
+            addGroupToLogs(type.toLowerCase());
+        }
+
+        return logs;
+    }
+
+    /**
+     * Print logs from the cache, to the console.
+     *
+     * @public
+     * @param {LOG_TYPES|LOG_TYPES[]} [type] - Type of logs to print. If empty, will print all logs.
+     * If a list is given, will print each entry specified.
+     * @return {void}
+     */
+    printLogs(type) {
+        const logs = this.getLogs(type);
```

Comment:
This would be used in situations where you want to print the entire history of logs/errors etc.
Since we won't be printing warnings or logs, this is how you would access them.